mac80211: update to version 6.18

Ansuel opened the pull request:
Backport support for faux base driver needed for backports package to
compile with kernel 6.18.

Drop all upstreamed patch and refresh all affected patch.

Minor changes were needed to the
350-mac80211-allow-scanning-while-on-radar-channel due to changes in
6.18.

Ansuel committed:
generic: backport support for faux base driver

Backport support for faux base driver needed for backports package to
compile with kernel 6.18.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
mac80211: update to version 6.18

Drop all upstreamed patch and refresh all affected patch.

Minor changes were needed to the
350-mac80211-allow-scanning-while-on-radar-channel due to changes in
6.18.

Also some changes were needed to downstream patch due to upstream commit
b74947b4f6ff ("wifi: cfg80211/mac80211: Add support to get radio index")

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

neheb commented:
btw 609-rt2x00-make-wmac-loadable-via-OF-on-rt288x-305x-SoC.patch needs to come back. Unless you add a migration script.

edit: more context: https://github.com/openwrt/openwrt/commit/d2b2670937f051bda7c8fbda8a6c0131977d5e68

Ansuel committed:
ramips: migrate wifi compatible to new name

Migrate wifi compatible from ralink,rt2880-wmac to ralink,rt2880-wifi to
align to new 6.18 backports version.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@neheb thanks for noticing that. We  dodged a bullet :D

neheb commented on target/linux/ramips/dts/mt7620a.dtsi:
```diff
@@ -627,7 +627,7 @@
 	};
 
 	wmac: wmac@10180000 {
-		compatible = "ralink,rt7620-wmac", "ralink,rt2880-wmac";
```
even though the first compatibles are completely unused, wouldn't it make sense to change those as well?

DragonBluep commented on target/linux/ramips/dts/mt7620a.dtsi:
```diff
@@ -627,7 +627,7 @@
 
 	wmac: wmac@10180000 {
-		compatible = "ralink,rt7620-wmac", "ralink,rt2880-wmac";
+		compatible = "ralink,rt7620-wifi", "ralink,rt2880-wifi";
```
The first compatible is used by the iwinfo to match the NIC model. If you want to change it, please also update the iwinfo package.

Ansuel replied:
```
"qca,ar9130-wmac"       0      0  "Atheros"  "AR9130"
"qca,ar9330-wmac"       0      0  "Atheros"  "AR9330"
"qca,ar9340-wmac"       0      0  "Atheros"  "AR9340"
"qca,qca9530-wmac"      0      0  "Qualcomm Atheros"  "QCA9530"
"qca,qca9550-wmac"      0      0  "Qualcomm Atheros"  "QCA9550"
"qca,qca9560-wmac"      0      0  "Qualcomm Atheros"  "QCA9560"
"qcom,ipq4019-wifi"     0      0  "Qualcomm Atheros" "IPQ4019"
"qcom,ipq5018-wifi"     0      0  "Qualcomm Atheros" "IPQ5018"
"qcom,ipq6018-wifi"     0      0  "Qualcomm Atheros" "IPQ6018"
"qcom,ipq8074-wifi"     0      0  "Qualcomm Atheros" "IPQ8074"
"qcom,qcn6122-wifi"     0      0  "Qualcomm Atheros" "QCN6102/QCN6122"
"mediatek,mt7622-wmac"  0      0  "MediaTek" "MT7622"
"mediatek,mt7628-wmac"  0      0  "MediaTek" "MT7628"
"mediatek,mt7981-wmac"  0      0  "MediaTek" "MT7981"
"mediatek,mt7986-wmac"  0      0  "MediaTek" "MT7986"
"ralink,rt2880-wmac"    0      0  "Ralink" "Rt2880"
"ralink,rt3050-wmac"    0      0  "Ralink" "Rt3050"
"ralink,rt3352-wmac"    0      0  "Ralink" "Rt3352"
"ralink,rt3883-wmac"    0      0  "Ralink" "Rt3883"
"ralink,rt5350-wmac"    0      0  "Ralink" "Rt5350"
"ralink,rt7620-wmac"    0      0  "MediaTek" "MT7620"
```

just notice also the other entry are outdated...

neheb replied:
wifi-scripts too

```
git grep \\\-wmac
package/kernel/mac80211/patches/rt2x00/609-rt2x00-make-wmac-loadable-via-OF-on-rt288x-305x-SoC.patch:+  { .compatible = "ralink,rt2880-wmac" },
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "mediatek,mt7622-wmac": [ "MediaTek", "MT7622" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "mediatek,mt7628-wmac": [ "MediaTek", "MT7628" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "mediatek,mt7981-wmac": [ "MediaTek", "MT7981" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "mediatek,mt7986-wmac": [ "MediaTek", "MT7986" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "ralink,rt2880-wmac": [ "Ralink", "Rt2880" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "ralink,rt3050-wmac": [ "Ralink", "Rt3050" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "ralink,rt3352-wmac": [ "Ralink", "Rt3352" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "ralink,rt3883-wmac": [ "Ralink", "Rt3883" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "ralink,rt5350-wmac": [ "Ralink", "Rt5350" ],
package/network/config/wifi-scripts/files-ucode/usr/share/wifi_devices.json:            "ralink,rt7620-wmac": [ "MediaTek", "MT7620" ]
target/linux/mediatek/patches-6.12/117-complete-mt7981b-dtsi.patch:                     compatible = "mediatek,mt7981-wmac";
target/linux/mediatek/patches-6.12/941-arm64-dts-mt7986-move-cpuboot-in-a-dedicated-node.patch:                         compatible = "mediatek,mt7986-wmac";
target/linux/ramips/dts/mt7620a.dtsi:           compatible = "ralink,rt7620-wmac", "ralink,rt2880-wmac";
target/linux/ramips/dts/mt7620n.dtsi:           compatible = "ralink,rt7620-wmac", "ralink,rt2880-wmac";
target/linux/ramips/dts/mt7628an.dtsi:          compatible = "mediatek,mt7628-wmac";
target/linux/ramips/dts/rt2880.dtsi:            compatible = "ralink,rt2880-wmac";
target/linux/ramips/dts/rt3050.dtsi:            compatible = "ralink,rt3050-wmac", "ralink,rt2880-wmac";
target/linux/ramips/dts/rt3352.dtsi:            compatible = "ralink,rt3352-wmac", "ralink,rt2880-wmac";
target/linux/ramips/dts/rt3883.dtsi:            compatible = "ralink,rt3883-wmac", "ralink,rt2880-wmac";
target/linux/ramips/dts/rt5350.dtsi:            compatible = "ralink,rt5350-wmac", "ralink,rt2880-wmac";
```

neheb replied:
ah yes. The common practice of breaking ABI and hoping Linus doesn't notice :).

neheb commented:
Since mac80211 is getting updated, are these relevant?

https://github.com/neheb/openwrt/commit/b93c8b4cd64e0c9039dfe661c2ccca9f6b7e9d62
https://github.com/neheb/openwrt/commit/0135e331d5bbe5801d4f00ae89a1b008cc5ed28f
https://github.com/neheb/openwrt/commit/7049165fec88be39f5fef104215f124d06609ec1

AgustinLorenzo commented:
Hi @Ansuel 

You can remove the patch “940-ath11k-clear_initialized_flag_for_deinit-ed_srng_lists.patch”, it is upstreamed.

Link (6.18-rc7): https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/net/wireless/ath/ath11k?h=v6.18-rc7&id=a5b46aa7cf5f05c213316a018e49a8e086efd98e

Link (6.16.12): https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/net/wireless/ath/ath11k?h=v6.18-rc7&id=a5b46aa7cf5f05c213316a018e49a8e086efd98e

Thanks.

Ansuel commented:
@AgustinLorenzo thanks for noticing that... somehow the patch was applying anyway... magic of fuzzing....

Borromini commented:
@gingernut1978 @EPinci Looks like the mt76 package might need updating? Does the build complete if you try to leave mt76 out? (Not sure if the buildroot allows you to without too much hacking.)

Ansuel committed:
mt76: backport patch fixing support for 6.18

Backport upstream patch fixing support for 6.18 due to kernel upstream
commit b74947b4f6ff ("wifi: cfg80211/mac80211: Add support to get radio
index")

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

gingernut1978 commented:
@Ansuel 

Latest commits seem to cause this error for me now

```
812-b43-add-antenna-control.patch using plaintext: 
patching file drivers/net/wireless/broadcom/b43/main.c
patch: **** malformed patch at line 45: @@ -5178,6 +5175,47 @@ static int b43_op_get_survey(struct ieee
```

Ansuel committed:
mwlwifi: add pending patch to support kernel 6.18

Add pending patch to support backports package based on kernel 6.18.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
generic: move reserved_mem backports patch to generic

Move reserved_mem backports patch from airoha to generic dir as it will
be also used by backports package based on 6.18.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

gingernut1978 commented:
@Ansuel any idea what could be causing this trace, happend before 6.18 backport update but seems it's from mac80211

mt7622 target

```
[2 dic 2025, 6:33:31 CET] kern.warn: [   19.973837] WARNING: CPU: 0 PID: 2025 at ieee80211_hw_conf_init+0x7c/0x84 [mac80211]
[2 dic 2025, 6:33:31 CET] kern.debug: [   19.981725] Modules linked in: pppoe ppp_async nft_fib_inet nf_flow_table_inet pppox ppp_generic nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject nft_redir nft_quota nft_numgen nft_nat nft_masq nft_log nft_limit nft_hash nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_ct nft_chain_nat nf_tables nf_nat nf_flow_table nf_conntrack mt7615e(O) mt7615_common(O) mt76_connac_lib(O) mt76(O) mac80211(O) cfg80211(O) ums_usbat ums_sddr55 ums_sddr09 ums_karma ums_jumpshot ums_isd200 ums_freecom ums_datafab ums_cypress ums_alauda slhc nfnetlink nf_reject_ipv6 nf_reject_ipv4 nf_log_syslog nf_defrag_ipv6 nf_defrag_ipv4 libcrc32c compat(O) cls_flower act_vlan fuse cls_bpf act_bpf sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact cifs oid_registry nls_ucs2_utils cifs_md4 cifs_arc4 asn1_decoder dns_resolver netfs sha512_arm64 seqiv md5 geniv des_generic libdes uas usb_storage leds_gpio xhci_plat_hcd xhci_pci xhci_mtk_hcd xhci_hcd
[2 dic 2025, 6:33:31 CET] kern.debug: [   19.982438]  gpio_button_hotplug(O) usbcore usb_common
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.076911] CPU: 0 UID: 101 PID: 2025 Comm: hostapd Tainted: G           O       6.12.60 #0
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.085258] Tainted: [O]=OOT_MODULE
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.088737] Hardware name: TOTOLINK A8000RU (DT)
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.093346] pstate: 60400005 (nZCv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.100301] pc : ieee80211_hw_conf_init+0x7c/0x84 [mac80211]
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.105986] lr : ieee80211_hw_conf_init+0x38/0x84 [mac80211]
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.111669] sp : ffffffc0818abaa0
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.114974] x29: ffffffc0818abaa0 x28: ffffff8000a94600 x27: ffffff8005304a00
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.122108] x26: ffffff8005a39940 x25: 0000000000000001 x24: ffffff8005a38000
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.129241] x23: ffffff8000360900 x22: ffffff8005a38950 x21: 0000000000000001
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.136375] x20: ffffff8005a38940 x19: ffffff8000360900 x18: 00000000ffffffff
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.143508] x17: 0000000000000028 x16: 0000000000000000 x15: 0000000000000230
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.150641] x14: ffffff80029e50e0 x13: ffffff8000a946c0 x12: 0000000000000000
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.157774] x11: 00000000000000c0 x10: 00000000000008e0 x9 : 0000000000000000
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.164906] x8 : ffffff801fea3ac0 x7 : 0000000000000001 x6 : 0000000000000002
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.172040] x5 : ffffff800278b5c0 x4 : 0000000000000000 x3 : ffffff8000a94600
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.179173] x2 : 0000000000000000 x1 : ffffff8000a94600 x0 : 00000000ffffffea
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.186306] Call trace:
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.188744]  ieee80211_hw_conf_init+0x7c/0x84 [mac80211]
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.194080]  ieee80211_do_open+0x580/0x740 [mac80211]
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.199156]  ieee80211_do_open+0x724/0x740 [mac80211]
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.204232]  __dev_open+0xe8/0x160
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.207633]  __dev_change_flags+0x154/0x1c0
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.211811]  dev_change_flags+0x20/0x64
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.215642]  devinet_ioctl+0x320/0x5e0
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.219387]  inet_ioctl+0x220/0x268
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.222871]  sock_ioctl+0x2f4/0x450
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.226357]  __arm64_sys_ioctl+0x354/0xe60
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.230449]  invoke_syscall.constprop.0+0x4c/0xd0
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.235149]  do_el0_svc+0x3c/0xd0
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.238457]  el0_svc+0x18/0x60
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.241508]  el0t_64_sync_handler+0x118/0x124
[2 dic 2025, 6:33:31 CET] kern.debug: [   20.245858]  el0t_64_sync+0x150/0x154
```

Ansuel committed:
rtl8812au-ct: rework patch to support kernel 6.18

Rework patch to support backports package based on kernel 6.18.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ath10k-ct: update to Git HEAD (2025-12-01)

bb84e1597474 Add 6.18 ath10k-ct driver

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
ath10k-ct: switch to base 6.18

Switch ath10k-ct kernel version base to 6.18 to align with backports
package based on 6.18 kernel version.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@gingernut1978 if it's unrelated to this update please lets not confuse things up :(

Ansuel commented:
Ok good waiting for ci to finish and will consider merging...

hauke commented on package/kernel/mac80211/patches/rt2x00/100-rt2x00_options.patch:
```diff
@@ -1,47 +1,11 @@
-+	tristate "RT2x00 support"
- 	depends on m
- 
- config RT2X00_LIB_FIRMWARE
```
This is still needed.
When we allow selecting this in a kmod which does not have `HIDDEN:=1` the Kconfig text needs a title to allow selecting it. `kmod-rt2x00-lib` is not hidden, some other are hidden and the title is not needed there any more.

robimarko commented:
@Ansuel This breaks ipq807x:

```
[   10.824055] ath11k c000000.wifi: ipq8074 hw2.0
[   10.824090] ath11k c000000.wifi: FW memory mode: 0
[   10.869879] remoteproc remoteproc0: powering up cd00000.q6v5_wcss
[   10.870195] remoteproc remoteproc0: Booting fw image IPQ8074/q6_fw.mdt, size 668
[   11.870207] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   11.914237] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   11.921638] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   11.921677] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   12.074592] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   12.074592] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   12.074592] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   12.074592] 
[   12.074592] 
[   12.076241] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   12.098496] remoteproc remoteproc0: handling crash #1 in cd00000.q6v5_wcss
[   12.106647] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   12.139349] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   12.485392] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   22.245298] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   22.245351] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   22.255090] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   22.263652] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   22.266380] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   22.368633] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   22.368633] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   22.368633] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   22.368633] 
[   22.368633] 
[   22.370270] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   22.392532] remoteproc remoteproc0: handling crash #2 in cd00000.q6v5_wcss
[   22.400674] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   22.433050] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   22.778293] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   32.485306] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   32.485322] l11: disabling
[   32.491775] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   32.497608] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   32.506335] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   32.509062] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   32.611307] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   32.611307] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   32.611307] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   32.611307] 
[   32.611307] 
[   32.612946] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   32.635205] remoteproc remoteproc0: handling crash #3 in cd00000.q6v5_wcss
[   32.643354] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   32.675824] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   33.021536] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   42.725297] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   42.725349] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   42.735072] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   42.743669] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   42.746375] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   42.847988] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   42.847988] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   42.847988] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   42.847988] 
[   42.847988] 
[   42.849623] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   42.871887] remoteproc remoteproc0: handling crash #4 in cd00000.q6v5_wcss
[   42.880032] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   42.912527] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   43.258265] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   52.965300] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   52.965352] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   52.975091] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   52.983656] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   52.986385] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   53.088352] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   53.088352] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   53.088352] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   53.088352] 
[   53.088352] 
[   53.089989] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   53.112245] remoteproc remoteproc0: handling crash #5 in cd00000.q6v5_wcss
[   53.120392] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   53.152741] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   53.497934] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   63.205296] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   63.205348] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   63.215072] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   63.223660] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   63.226374] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   63.328676] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   63.328676] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   63.328676] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   63.328676] 
[   63.328676] 
[   63.330314] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   63.352574] remoteproc remoteproc0: handling crash #6 in cd00000.q6v5_wcss
[   63.360723] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   63.393249] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   63.737836] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   73.445297] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   73.445349] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   73.455079] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   73.463664] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   73.466379] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   73.569087] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   73.569087] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   73.569087] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   73.569087] 
[   73.569087] 
[   73.570723] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   73.592986] remoteproc remoteproc0: handling crash #7 in cd00000.q6v5_wcss
[   73.601140] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   73.633643] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   73.979451] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   74.725303] ath11k c000000.wifi: Coldboot Calibration timed out
[   74.728457] kmodloader: done loading kernel modules from /etc/modules.d/*
[   77.338957] br-lan: port 1(lan1) entered blocking state
[   77.339009] br-lan: port 1(lan1) entered disabled state
[   77.343043] nss-dp 3a001600.dp4 lan1: entered allmulticast mode
[   77.348461] nss-dp 3a001600.dp4 lan1: entered promiscuous mode
[   77.357289] br-lan: port 2(lan2) entered blocking state
[   77.360025] br-lan: port 2(lan2) entered disabled state
[   77.365194] nss-dp 3a001400.dp3 lan2: entered allmulticast mode
[   77.370605] nss-dp 3a001400.dp3 lan2: entered promiscuous mode
[   77.378572] br-lan: port 3(lan3) entered blocking state
[   77.382158] br-lan: port 3(lan3) entered disabled state
[   77.387392] nss-dp 3a001200.dp2 lan3: entered allmulticast mode
[   77.392682] nss-dp 3a001200.dp2 lan3: entered promiscuous mode
[   77.400932] br-lan: port 4(lan4) entered blocking state
[   77.404294] br-lan: port 4(lan4) entered disabled state
[   77.409515] nss-dp 3a001000.dp1 lan4: entered allmulticast mode
[   77.414825] nss-dp 3a001000.dp1 lan4: entered promiscuous mode
[   77.425877] br-lan: port 5(wan) entered blocking state
[   77.426427] br-lan: port 5(wan) entered disabled state
[   77.431585] nss-dp 3a007000.dp6-syn wan: entered allmulticast mode
[   77.436980] nss-dp 3a007000.dp6-syn wan: entered promiscuous mode
[   81.605740] nss-dp 3a007000.dp6-syn wan: PHY Link up speed: 2500
[   81.606279] br-lan: port 5(wan) entered blocking state
[   81.610828] br-lan: port 5(wan) entered forwarding state
[   83.685315] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   83.685373] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   83.695117] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   83.703667] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   83.706399] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   83.808873] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   83.808873] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   83.808873] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   83.808873] 
[   83.808873] 
[   83.810511] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   83.832775] remoteproc remoteproc0: handling crash #8 in cd00000.q6v5_wcss
[   83.840919] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   83.873593] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   84.218781] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[   93.925310] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[   93.925365] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[   93.935115] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[   93.943675] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[   93.946391] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   94.048577] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[   94.048577] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[   94.048577] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[   94.048577] 
[   94.048577] 
[   94.050216] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[   94.072485] remoteproc remoteproc0: handling crash #9 in cd00000.q6v5_wcss
[   94.080621] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[   94.113145] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[   94.456955] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  104.165298] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  104.165351] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  104.175103] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  104.183655] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  104.186380] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  104.288605] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  104.288605] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  104.288605] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  104.288605] 
[  104.288605] 
[  104.290244] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  104.312515] remoteproc remoteproc0: handling crash #10 in cd00000.q6v5_wcss
[  104.320647] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  104.353064] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  104.698329] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  114.405307] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  114.405364] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  114.415111] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  114.423661] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  114.426392] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  114.529229] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  114.529229] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  114.529229] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  114.529229] 
[  114.529229] 
[  114.530865] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  114.553134] remoteproc remoteproc0: handling crash #11 in cd00000.q6v5_wcss
[  114.561275] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  114.593966] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  114.940055] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  124.645305] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  124.645359] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  124.655109] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  124.663663] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  124.666385] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  124.769296] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  124.769296] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  124.769296] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  124.769296] 
[  124.769296] 
[  124.770934] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  124.793198] remoteproc remoteproc0: handling crash #12 in cd00000.q6v5_wcss
[  124.801338] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  124.833752] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  125.177524] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  134.885321] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  134.885376] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  134.895126] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  134.903672] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  134.906404] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  135.008563] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  135.008563] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  135.008563] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  135.008563] 
[  135.008563] 
[  135.010200] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  135.032465] remoteproc remoteproc0: handling crash #13 in cd00000.q6v5_wcss
[  135.040611] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  135.073191] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  135.419119] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  145.125373] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  145.125429] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  145.135175] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  145.143740] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  145.146460] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  145.248951] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  145.248951] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  145.248951] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  145.248951] 
[  145.248951] 
[  145.250591] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  145.272858] remoteproc remoteproc0: handling crash #14 in cd00000.q6v5_wcss
[  145.281001] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  145.313602] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  145.658699] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  155.365363] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  155.365421] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  155.375169] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  155.383723] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  155.386448] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  155.489313] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  155.489313] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  155.489313] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  155.489313] 
[  155.489313] 
[  155.490949] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  155.513212] remoteproc remoteproc0: handling crash #15 in cd00000.q6v5_wcss
[  155.521359] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  155.554011] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  155.899294] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
[  165.605383] ath11k c000000.wifi: failed to wait wlan mode request (mode 7): -110
[  165.605437] ath11k c000000.wifi: qmi failed to send wlan fw mode: -110
[  165.615218] ath11k c000000.wifi: qmi fail to get qcom,m3-dump-addr, ignore m3 dump mem req
[  165.623749] ath11k c000000.wifi: chip_id 0x0 chip_family 0x0 board_id 0xff soc_id 0xffffffff
[  165.626464] ath11k c000000.wifi: fw_version 0x290b84a5 fw_build_timestamp 2024-09-23 11:32 fw_build_id WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  165.729109] qcom-q6v5-wcss-pil cd00000.q6v5_wcss: fatal error received: 
[  165.729109] QC Image Version: QC_IMAGE_VERSION_STRING=WLAN.HK.2.9.0.1-02146-QCAHKSWPL_SILICONZ-1
[  165.729109] Image Variant : IMAGE_VARIANT_STRING=8074.wlanfw.eval_v2Q
[  165.729109] 
[  165.729109] 
[  165.730747] remoteproc remoteproc0: crash detected in cd00000.q6v5_wcss: type fatal error
[  165.753011] remoteproc remoteproc0: handling crash #16 in cd00000.q6v5_wcss
[  165.761158] remoteproc remoteproc0: recovering cd00000.q6v5_wcss
[  165.793815] remoteproc remoteproc0: stopped remote processor cd00000.q6v5_wcss
[  166.139819] remoteproc remoteproc0: remote processor cd00000.q6v5_wcss is now up
```

danpawlik commented:
Got some difficulties on Airoha (Quantum Fiber W1700K):
```
-- Generating done (0.0s)
-- Build files have been written to: /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/tools
touch /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/.configured_b1c6ae92f8e974e60b47b3e5ad3573cc
rm -f /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/.built
touch /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/.built_check
make -C /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/linux-6.12.59 KCFLAGS="-ffile-prefix-map=/home/user/openwrt-builder/
openwrt/build_dir/target-aarch64_cortex-a53_musl=target-aarch64_cortex-a53_musl -fno-caller-saves " HOSTCFLAGS="-O2 -I/home/user/openwrt-builder/openwrt/staging_dir/host/in
clude -I/home/user/openwrt-builder/openwrt/staging_dir/hostpkg/include -I/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/host/include -Wall -W
missing-prototypes -Wstrict-prototypes" CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_T
IMESTAMP="Tue Dec  2 20:32:16 2025" KBUILD_BUILD_VERSION="0" KBUILD_HOSTLDFLAGS="-L/home/user/openwrt-builder/openwrt/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_sy
scalls= KBUILD_EXTRA_SYMBOLS="/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/symvers/cryptodev-linux.symvers /home/user/ope
nwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/symvers/gpio-button-hotplug.symvers /home/user/openwrt-builder/openwrt/build_dir/target-aa
rch64_cortex-a53_musl/linux-airoha_an7581/symvers/mac80211.symvers" CC="aarch64-openwrt-linux-musl-gcc" KERNELRELEASE=6.12.59  CONFIG_MT76_CONNAC_LIB=m CONFIG_MT7996E=m CON
FIG_MT76_NPU=y CONFIG_MT7996_NPU=y M="/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7" NOSTDINC_FLA
GS="-nostdinc  -isystem /home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-15.2.0_musl/lib/gcc/aarch64-openwrt-linux-musl/15.2.0/include -I/hom
e/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7 -I/home/user/openwrt-builder/openwrt/staging_dir/target
-aarch64_cortex-a53_musl/usr/include/mac80211-backport/uapi -I/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/mac80211-backport -I
/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/mac80211/uapi -I/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cort
ex-a53_musl/usr/include/mac80211 -include backport/autoconf.h -include backport/backport.h -DCONFIG_MAC80211_MESH -DCONFIG_MT76_NPU -DCONFIG_MT7996_NPU" modules
make[3]: Entering directory '/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/linux-6.12.59'
  CC [M]  /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/mmio.o
In file included from <command-line>:
././include/linux/compiler_types.h:339:10: error: '__counted_by' redefined [-Werror]
  339 | # define __counted_by(member)           __attribute__((__counted_by__(member)))
      |          ^~~~~~~~~~~~
In file included from ././include/linux/compiler_types.h:89:
/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/mac80211-backport/linux/compiler_attributes.h:45:10: note: this is the location of
 the previous definition
   45 | # define __counted_by(member)
      |          ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make[5]: *** [scripts/Makefile.build:229: /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/mmio.o] E
rror 1
make[4]: *** [/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/linux-6.12.59/Makefile:1945: /home/user/openwrt-builder/openwr
t/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7] Error 2
make[3]: *** [Makefile:224: __sub-make] Error 2
make[3]: Leaving directory '/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/linux-6.12.59'
make[2]: *** [Makefile:732: /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-airoha_an7581/mt76-2025.11.06~eb567bc7/.built] Error 2
make[2]: Leaving directory '/home/user/openwrt-builder/openwrt/package/kernel/mt76'
time: package/kernel/mt76/compile#1.04#0.64#1.73
    ERROR: package/kernel/mt76 failed to build.
make[1]: *** [package/Makefile:187: package/kernel/mt76/compile] Error 1
make[1]: Leaving directory '/home/user/openwrt-builder/openwrt'
make: *** [/home/user/openwrt-builder/openwrt/include/toplevel.mk:233: package/kernel/mt76/compile] Error 2
```

Ansuel committed:
generic: add pending patch fixing resource_size()

Add opending patch fixing resource_size(9 if resource start and end is
set to zero.

This is to indirectly fix a helper rework in ath11k qmi code that cause
firmware load panic.

Link: https://github.com/openwrt/openwrt/pull/20964
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

EPinci commented:
Yeah, I tried to back trace it to a commit before this and I get as well - thou less frequently. This upgrade is not the cause.
